lib/test_prof/recipes/minitest/before_all.rb

```diff
@@ -2,6 +2,21 @@
 
 require "test_prof/before_all"
 
+Minitest.singleton_class.prepend(Module.new do
+  @previous_klass = nil
+
+  def run_one_method(klass, method_name)
+    return super unless klass.parallelized
+
+    if @previous_klass && @previous_klass != klass
+      @previous_klass.before_all_executor&.deactivate!
```

Comment:
So, we call `deactivate!` every time a new test class is encountered. But how do we call it for the last test class?

Reply:
If we find a way to perform the last `deactivate!`, we could probably get rid of all the patches and only rely on this logic for both parallelized and non-parallelized scenarios (as far as I understand).

Reply:
That's a fair point! I've added a `parallelize_teardown` hook to clean up the last transactions in my latest commit.

As far as using this logic for non-parallelized tests as well: That's an interesting idea. I can certainly look into which other patches we can drop. But since `parallelize_teardown` is not called for non-parallelized tests, we would have to keep at least some of the existing patches around I gues.

Reply:
Could you please take a look at test failures?

Reply:
Alright, test cases should be fixed.
While doing that, I also realized that this will probably only work in Rails, not with pure Minitest. I.e. it depends on the tests inheriting from `ActiveSupport::TestCase`. Is that an issue?

Reply:
Let's add a check for the presence of the `parallelize` method and we should be good to go (we must do that anyway since we use `super` in the prepended module).

Thus, if no `parallelize` method defined, we do nothing.

Reply:
Done 👍